### PR TITLE
UI Plugin: add management cluster creation for the docker workflow

### PR DESCRIPTION
**Pull request by raymondz1**
## What this PR does / why we need it

finished management cluster creation for docker workflow.

## Which issue(s) this PR fixes

Fixes: https://github.com/vmware-tanzu/community-edition/issues/4508

### CODE REVIEW

File: cli/cmd/plugin/ui/web/tanzu-ui/node-server/src/routes/api/endpoints/docker.js [hunk 1]
```diff
@@ -53,7 +51,8 @@ router.post('/tkgconfig', (req, res) => {
  */
 router.post('/config/export', (req, res) => {
     winston.info('Mock UI export config');
-    res.status(200);    res.status(200);
+    res.status(200);
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/node-server/src/routes/api/endpoints/docker.js:54]: looks like duplicate response status

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/services/awsDeployment.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                description: '',
+                location: '',
+            },
+            identityManagement: {
+                idm_type: IdentityManagementConfig.idm_type.NONE,
+            },
+        };
+        return awsClusterParams;
+    };
+    const deployOnAws = async () => {
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/services/awsDeployment.tsx:88]: nitpick, add a line break here

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/services/awsDeployment.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            dispatch({
+                type: DEPLOYMENT_STATUS_CHANGED,
+                payload: {
+                    type: DeploymentTypes.MANAGEMENT_CLUSTER,
+                    status: DeploymentStates.RUNNING,
+                    provider: Providers.AWS,
+                    configPath: configFileInfo.path,
+                },
+            });
+        } catch (e) {
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/services/awsDeployment.tsx:104]: please change the console output to something like:

`Error when calling config or create API: ${e}`

File: cli/cmd/plugin/ui/web/tanzu-ui/src/components/aws/AwsMCCreateSimple.tsx [hunk 4]
```diff
@@ -4,20 +4,23 @@ import React, { useContext } from 'react';
 // App imports
 import { AwsStore } from '../../state-management/stores/Store.aws';
 import { TAB_NAMES } from '../../shared/constants/NavRoutes.constants';
-import ManagementClusterSettings from './management-cluster-settings-step/ManagementClusterSettings';
-import ManagementCredentials from './management-credential-step/ManagementCredentials';
 import Wizard from '../../shared/components/wizard/Wizard';
+import ManagementCredentials from './management-credential-step/ManagementCredentials';
+import ManagementClusterSettings from '../../shared/components/management-cluster-settings-step/ManagementClusterSettings';
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/components/aws/AwsMCCreateSimple.tsx:9]: nit: ManagementCluster comes before ManagementCredentials

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/services/awsDeployment.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+import { Providers } from '../constants/Providers.constants';
+
+const useAwsDeployment = () => {
+    const { dispatch } = useContext(Store);
+    const { awsState } = useContext(AwsStore);
+    const navigate = useNavigate();
+    const navigateToProgress = (): void => {
+        navigate('/' + NavRoutes.DEPLOY_PROGRESS);
+    };
+    const getAwsRequestPayload = () => {
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/services/awsDeployment.tsx:30]: please add line break here

**Review by swalner-vmware — APPROVED** [review]

File: cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    deploy: () => void;
+    defaultData?: {[key:string]: any}
+}
+
+const nodeProfiles = [
+    {
+        label: 'Single node',
+        icon: 'block',
+        message:
+            'Create one control plane-node with a general purpose instance type in a single region.',
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/management-cluster-settings-step/ManagementClusterSettings.tsx:40]: sorry should not be 'plane-node'

should instead be a space between

**Review by miclettej — APPROVED** [review]

**Pull request merged**